internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/integration/utils/VeniceServerWrapper.java

```diff
@@ -225,28 +229,38 @@ static StatefulServiceProvider<VeniceServerWrapper> generateService(
       // generate the kafka cluster map in config directory
       VeniceConfigLoader.storeKafkaClusterMap(configDirectory, kafkaClusterMap);
 
-      if (!forkServer) {
-        VeniceConfigLoader veniceConfigLoader =
-            VeniceConfigLoader.loadFromConfigDirectory(configDirectory.getAbsolutePath());
+      String d2ServiceName = D2TestUtils.getRouterD2ServiceName(null, clusterName);
```

Comment:
Using this function is okay but let's change the function name to say `getRandomD2ServiceName`

Reply:
I changed the approach here to use `clusterName + "_test"` as the `d2ServiceName`. The integration test also needed the `d2ServiceName` and I couldn't find a way to pull it so I'm kind of hardcoding it. Is this fine?

Reply:
Let's find a different way. Maybe `VeniceServerWrapper.getD2ServiceName` or `VeniceClusterWrapper.getServerD2ServiceName()`?

Reply:
I'm not sure a getter will work because the method is static and we can't set the `VeniceServerWrapper`'s `d2ServiceName` field from it.

However, I noticed the `d2ServiceName` is set only if a `consumerClientConfig` is provided in the constructor. I tried to pass a config in the `featureProperties` of `cluster.addVeniceServer()` in my test and then retrieve the `d2ServiceName` from the `VeniceServerWrapper`. But this approach feels like rabbit hole, I can't get it to work.